[sun-ruiheng] iP

sun-ruiheng opened the pull request:
## GrumbleBug
> A grumpy descendant of Duke... best left alone.

GrumbleBug is a simple robot that helps you track tasks. He also hates talking to people.

What GrumbleBug likes: 
- [pizza](https://www.dominos.com.sg/)
- being left alone 😬

What GrumbleBug dislikes:
- Talking
- People 
- Talking to people 🙁

GrumbleBug helps to...
- [ ] Enter and delete tasks
- [ ] Mark and unmark tasks as done
- [ ] Specify types and descriptions of tasks
- [ ] Store all this information in its `taskList`
- [ ] Communicate with local storage through its `storage`
- [ ] Finish your leftover pizza

Generally, you can follow these steps:
1. Open up GrumbleBug
2. Add some tasks and their corresponding information
3. When you complete a task, tell GrumbleBug to mark it as done
4. Check out your task list to see your progress!

### Let's have a sneak peek at how GrumbleBug is built!
```java
    public GrumbleBug() {
        this.storage = new Storage();
        this.taskList = new TaskList();
        this.ui = new Ui();
        this.parserInput = new Parser("yyyy-MM-dd");
    }
```
_Awesome, he follows nice OOP principles!_

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

sun-ruiheng committed:
First commit, L0

sun-ruiheng committed:
Add echo dialogue

sun-ruiheng committed:
Add list feature

sun-ruiheng committed:
add doneness to list items

sun-ruiheng committed:
Overloading for Task.java

sun-ruiheng committed:
Introduce types of event

sun-ruiheng committed:
Better behavior for level 4

sun-ruiheng committed:
Even Better behavior for level 4

sun-ruiheng committed:
Auto test

sun-ruiheng committed:
Add some error handlers

sun-ruiheng committed:
Delete option

sun-ruiheng committed:
fixed a small mistake from week 2

sun-ruiheng committed:
Add file fetching

sun-ruiheng committed:
Save and load functionality

sun-ruiheng committed:
Level 7 retry

sun-ruiheng committed:
Level 7 try branch

sun-ruiheng committed:
Merge branch 'branch-Level-7'

sun-ruiheng committed:
Add spaces

sun-ruiheng committed:
Merge branch 'branch-Level-7'

sun-ruiheng committed:
Level 8 datetime

sun-ruiheng committed:
Merge branch 'branch-Level-8'

sun-ruiheng committed:
Make more OOP

sun-ruiheng committed:
Move into grumblebug package

sun-ruiheng committed:
Merge branch 'local-add-gradle-support'

sun-ruiheng committed:
Add gradle

sun-ruiheng committed:
Add JUnit tests

sun-ruiheng committed:
JAR file? Unsure

sun-ruiheng committed:
Add some JavaDoc

sun-ruiheng committed:
Make sure we follow coding standard

sun-ruiheng committed:
Add filtering functionality

sun-ruiheng committed:
Add more javadoc

sun-ruiheng committed:
Brush up coding standard

sun-ruiheng committed:
Interm commit, add some javafx

sun-ruiheng committed:
Add premature UI to GrumbleBug

casaarlai commented:
Shouldn't some of these items be marked as completed already?

> GrumbleBug helps to...
> 
> * [ ]  Enter and delete tasks
> * [ ]  Mark and unmark tasks as done
> * [ ]  Specify types and descriptions of tasks
> * [ ]  Store all this information in its `taskList`
> * [ ]  Communicate with local storage through its `storage`
> * [ ]  Finish your leftover pizza

casaarlai commented on tasks.txt:
Is there a reason the personal tasks file is committed?

casaarlai commented on src/main/java/grumblebug/DialogBox.java:
```diff
@@ -0,0 +1,47 @@
+import javafx.scene.layout.HBox;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.Node;
```
Should this be moved up so that the ordering of import statements is consistent as per the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#package-and-import-statements)?

casaarlai commented on src/main/java/grumblebug/Storage.java:
```diff
@@ -0,0 +1,82 @@
+        try {
+            File f = new File(filePath);
+            f.createNewFile();
+            Scanner s = new Scanner(f);
```
Should the variable name here for _s_ be longer and more descriptive especially since the scope it is used in is quite large?

> Variables with a large scope should have long names, variables with a small scope can have short names.

casaarlai commented on src/main/resources/images/nailong.jpg:
Cute

casaarlai commented on src/test/java/grumblebug/TaskListTest.java:
```diff
@@ -0,0 +1,26 @@
+
+public class TaskListTest {
+    @Test
+    public void addTest() {
```
Could the test functions be named in the three part format `featureUnderTest_testScenario_expectedBehavior()`?

sun-ruiheng committed:
Create jar file with gradle and standardise build.gradle file.

Following the textbook, I standardise the build.gradle file to utilise its functionality better, including creating fat jar files with shadowJar to capture external libraries like JavaFX. I also create a jar file that is executable.

This way, I make the building process easier and more standardised for future extension of this project. It is also easier for users or reviewers to run the jar file when it is pushed to remote repo.

sun-ruiheng committed:
Add assert statements at appropriate points to ensure certain assumptions made by the programmer are true.

This includes assertions in the TaskList and GrumbleBug files. They keep tabs on external classes and methods not implemented by this project such as StringBuilder and the .split String method, so we have no direct eyes on the details and we want to ensure we understand the endpoints' behavior correctly.

sun-ruiheng committed:
Merge pull request #1 from sun-ruiheng/branch-A-Assertions

Add assert statements

sun-ruiheng committed:
Modify code to improve Code Quality according to our textbook recommendations.

Some changes made include naming of variables to make them clearer: methods should sound like actions that tell you exactly what it does, and variables should sound like descriptive nouns. These are done by refactoring names.

There are also code structure changes, such as intuitive use of line spacings.

Code quality should be diligently maintained and frequently reviewed. There might be other improvements I can make that I did not manage to spot this time, but I hope this is a good step in the right direction.

sun-ruiheng committed:
Merge pull request #2 from sun-ruiheng/branch-A-CodeQuality

Modify code to improve code quality

sun-ruiheng committed:
Add streams functionality in TaskList to make things more fluid and efficient.

sun-ruiheng committed:
Merge pull request #3 from sun-ruiheng/branch-A-Streams

Add streams functionality in TaskList

sun-ruiheng committed:
Add more personality

sun-ruiheng committed:
Add more JavaDoc and more respect for SLAP.

sun-ruiheng committed:
Fix some bugs

sun-ruiheng committed:
Restore saving and loading feature

sun-ruiheng committed:
Add user guide, fix SLAP problems

sun-ruiheng committed:
Update Parser terminate

sun-ruiheng committed:
fix README

sun-ruiheng committed:
Remove prints

sun-ruiheng committed:
Fix bugs

sun-ruiheng committed:
Fix bugs

sun-ruiheng committed:
Fix last bugs

sun-ruiheng committed:
Fix bugs according to reviews

sun-ruiheng committed:
Merge pull request #6 from sun-ruiheng/branch-touchup-OOP

Fix bugs according to reviews
